src/lib/mina/fetch.unit-test.ts

```diff
@@ -136,96 +137,1181 @@ describe('Fetch', async (t) => {
+      );
+
+      test('orders the actions correctly', () => {
+        expect(actionsList).toEqual([
+          {
+            actions: [
+              [
+                '20374659537065244088703638031937922870146667362923279084491778322749365537089',
+                '1',
+              ],
+            ],
+            hash: '10619825168606131449407092474314250900469658818945385329390497057469974757422',
+          },
+          {
+            actions: [
+              [
+                '20503089751358270987184701275168489753952341816059774976784079526478451099801',
+                '1',
+              ],
+            ],
+            hash: '25525130517416993227046681664758665799110129890808721833148757111140891481208',
+          },
+          {
+            actions: [
+              [
+                '3374074164183544078218789545772953663729921088152354292852793744356608231707',
+                '0',
+              ],
+            ],
+            hash: '290963518424616502946790040851348455652296009700336010663574777600482385855',
+          },
+          {
+            actions: [
+              [
+                '12630758077588166643924428865613845067150916064939816120404808842510620524633',
+                '1',
+              ],
+            ],
+            hash: '20673199655841577810393943638910551364027795297920791498278816237738641857371',
+          },
+          {
+            actions: [
+              [
+                '5643224648393140391519847064914429159616501351124129591669928700148350171602',
+                '0',
+              ],
+            ],
+            hash: '5284016523143033193387918577616839424871122381326995145988133445906503263869',
+          },
+          {
+            actions: [
+              [
+                '15789351988619560045401465240113496854401074115453702466673859303925517061263',
+                '0',
+              ],
+            ],
+            hash: '16944163018367910067334012882171366051616125936127175065464614786387687317044',
+          },
+          {
+            actions: [
+              [
+                '27263309408256888453299195755797013857604561285332380691270111409680109142128',
+                '1',
+              ],
+            ],
+            hash: '23662159967366296714544063539035629952291787828104373633198732070740691309118',
+          },
+          {
+            actions: [
+              [
+                '3378367318331499715304980508337843233019278703665446829424824679144818589558',
+                '1',
+              ],
+            ],
+            hash: '1589729766029695153975344283092689798747741638003354620355672853210932754595',
+          },
+          {
+            actions: [
+              [
+                '17137397755795687855356639427474789131368991089558570411893673365904353943290',
+                '1',
+              ],
+            ],
+            hash: '10964420428484427410756859799314206378989718180435238943573393516522086219419',
```

Comment:
given our history with the action sequence, I think we should seriously consider some advanced testing here - similar to the property testing we do in other places of the codebase

Reply:
Can you describe what cases you'd like to see?  The existing tests here are simply real payloads from the graphQL node, and the real output of `fetchActions` in a case where the actions are derived successfully.  In particular, it is one of the edge cases discovered by the original reporter of the issue: https://github.com/o1-labs/o1js/issues/1872#issue-2595537724

Reply:
Take a look at this https://github.com/o1-labs/o1js/blob/00eb12df2b35cd7bc6f020570e20c5e28eee8ed5/src/lib/testing/property.ts